core/src/main/java/org/apache/accumulo/core/util/threads/AccumuloUncaughtExceptionHandler.java

```diff
@@ -44,7 +46,10 @@ public void uncaughtException(Thread t, Throwable e) {
         // If e == OutOfMemoryError, then it's probably that another Error might be
         // thrown when trying to print to System.err.
       } finally {
-        Runtime.getRuntime().halt(-1);
+        Mode m = SingletonManager.getMode();
+        if (m != null && m.equals(Mode.SERVER)) {
+          Runtime.getRuntime().halt(-1);
+        }
```

Comment:
Rather than making this handler behave differently in some circumstances, why not make it so a different handler can be configured instead?

Reply:
Never mind on this. I think the current idea of providing the executor factory is better.